STM32/TransformationFunctions/Src/goertzel.c

```diff
@@ -2,63 +2,71 @@
 #include "math.h"
 
 static Goertzel hGoertzel;
+static float q0, q1, q2 = 0;
+static int sample_no = 0;
 
-// Initialize the Goertzel filter. 
-// Explanations of the parameters are found in goertzel.h
 /* Input Parameters:
 
-	targetFrequency: 	The frequency [Hz] of which the magnitude should be computed
+    adcres:				The maximum ADC value 12-bit: 4096, 16-bit: 65536, ...
+    vRange:				Voltage range of the ADC typically 3.3V
+	gain:				Any PCB specific gain from schematic. 
+    targetFrequency: 	The frequency [Hz] of which the magnitude should be computed
 	samplerate:			The samplerate [Hz] of pData
 	samplesPerOutput:	Total number of samples before the magnitude is computed
-	adcres:				The maximum ADC value 12-bit: 4096, 16-bit: 65536
-	vRange:				Voltage range of the ADC typically 3.3V
-	gain:				Any PCB specific gain from schematic. 
+	vToUnit:			Conversion factor from Voltage to other unit (bias independent)
 */ 
-void GoertzelInit(int targetFrequency, int sampleRate, float samplesPerOutput, float adcres, float vRange, float gain)
+void GoertzelInit(float adcres, float vRange, float gain, int targetFrequency, int sampleRate, float samplesPerOutput, float vToUnit)
```

Comment:
I must say that I don't have a particularly good reason. The goerzel in the libraries was copied a long time ago when it was first used in the AMB project. Since then some changes have been made to the function e.g. adding/removing of real and imaginary as output and the vToUnit parameter. 

I copied it from the AMB from now, so I didn't notice the shift. It isn't used elsewhere yet, so if alright with you I will leave it like this to be compliant with the amb project. If not, I will change the order back.